src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointInfo.cs

```diff
@@ -5,43 +5,43 @@
 using System.Collections.Generic;
 using System.Globalization;
 using Microsoft.AspNetCore.Routing;
-using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.AspNetCore.Routing.Patterns;
 
 namespace Microsoft.AspNetCore.Builder
 {
     internal class MvcEndpointInfo
     {
         public MvcEndpointInfo(
             string name,
-            string template,
+            string pattern,
             RouteValueDictionary defaults,
             IDictionary<string, object> constraints,
             RouteValueDictionary dataTokens,
             IInlineConstraintResolver constraintResolver)
         {
             Name = name;
-            Template = template ?? string.Empty;
+            Pattern = pattern ?? string.Empty;
             DataTokens = dataTokens;
 
             try
             {
-                // Data we parse from the template will be used to fill in the rest of the constraints or
+                // Data we parse from the pattern will be used to fill in the rest of the constraints or
                 // defaults. The parser will throw for invalid routes.
-                ParsedTemplate = TemplateParser.Parse(template);
+                ParsedPattern = RoutePatternFactory.Parse(pattern);
```

Comment:
Instead of MvcEndpointInfo doing the merging of defaults and constraints, I think we use the Parse overload which takes in defaults and constaints